examples/structured_data/feature_preprocessing_layer.py

```diff
@@ -0,0 +1,224 @@
+            self.categoric_string_lookup(data["categorical_features"])
+        )
+
+        self.discretizer.adapt(data["discretize_features"])
+
+        self.embedding_string_lookup.adapt(data["embedding_features"])
+
+        # create sequential model for string and integer lookup layers
+        self.categoric_encoding = tf.keras.models.Sequential(
+            [self.categoric_string_lookup, self.integer_lookup]
+        )
+
+        vocabulary_size = self.embedding_string_lookup.vocabulary_size()
+        embedding_size = int(tf.math.sqrt(vocabulary_size))
+        # create sequential model for embedding feature
+        self.embedding = tf.keras.models.Sequential(
+            [
+                self.embedding_string_lookup,
+                Embedding(input_dim=vocabulary_size, output_dim=embedding_size),
+                Reshape((-1,)),
+            ]
+        )
+
+    def call(self, inputs={}):
+        """Apply adapted layers"""
+        numerical_features = self.normalizer(inputs["numerical_features"])
+
+        categorical_features = self.categoric_encoding(inputs["categorical_features"])
+
+        discretize_features = tf.cast(
+            self.discretizer(inputs["discretize_features"]), dtype=tf.float32
+        )
+
+        embedding_features = self.embedding(inputs["embedding_features"])
+
+        return tf.keras.layers.concatenate(
+            [
+                numerical_features,
+                categorical_features,
+                discretize_features,
+                embedding_features,
+            ]
+        )
+
+
+"""
+## Compile and fit model
+
+We now have a feature preprocessing layer. Lets put everything together and fit the model:
+"""
+
+inputs = {
+    "numerical_features": tf.keras.Input(
+        shape=(len(NUMERIC_FEATURES),), dtype=tf.float32, name="numeric_layer"
+    ),
+    "categorical_features": tf.keras.Input(
+        shape=(len(CATEGORICAL_FEATURES),), dtype=tf.string, name="categoric_layer"
+    ),
+    "discretize_features": tf.keras.Input(
+        shape=(len(DISCRETIZE_FEATURES),), dtype=tf.float32, name="discretize_layer"
+    ),
+    "embedding_features": tf.keras.Input(
+        shape=(len(EMBEDDING_FEATURES),), dtype=tf.string, name="embedding_layer"
+    ),
+}
+
+# update preprocessing layer states
+preprocessing_layer = FeaturePreprocessor()
+preprocessing_layer.adapt(data_dict)
+
+# rest of model
+x = preprocessing_layer(inputs)
+x = tf.keras.layers.Dense(64, activation="relu")(x)
+x = tf.keras.layers.Dense(32, activation="relu")(x)
+outputs = tf.keras.layers.Dense(1, activation="sigmoid")(x)
+model = tf.keras.Model(inputs=inputs, outputs=outputs)
+
+model.compile(
+    loss=tf.keras.losses.BinaryCrossentropy(),
+    optimizer=tf.keras.optimizers.RMSprop(learning_rate=0.001),
+    metrics=[
+        tf.keras.metrics.BinaryAccuracy(name="accuracy"),
+        tf.keras.metrics.AUC(name="auc"),
+    ],
+)
+
+model.fit(data_dict, labels_binary.values, epochs=10, validation_split=0.25)
```

Comment:
There needs to be a section where we use the model to do inference on new data and we discuss test performance.